src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -392,6 +392,43 @@ public void setRsyncScript(String rsyncScript) {
     public String getRsyncScriptFilename() {
         return rsyncScriptFilename;
     }
+    
+    private Boolean hasValidTermsOfAccess = null;
+    
+    public Boolean isHasValidTermsOfAccess() {
+        //cache in page to limit processing
+        if (hasValidTermsOfAccess != null){
+            return hasValidTermsOfAccess;
+        } else {
+            if (!isHasRestrictedFiles()){
+               hasValidTermsOfAccess = true;
+               return hasValidTermsOfAccess;
+            } else {
+                hasValidTermsOfAccess = TermsOfUseAndAccessValidator.isTOUAValid(workingVersion.getTermsOfUseAndAccess(), null);
```

Comment:
It seems like the new validator is only called from the GUI? Are the same rules enforced via API?

Reply:
As I mentioned before we all left for break, the validator is called by the commands via the validateOrDie method so it will be applied to all dataset versions created or updated using the various apis.